### PR TITLE
Private/mmeeks/valgrindbits42

### PR DESCRIPTION
**Pull request by mmeeks**
* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: kit/Kit.cpp [hunk 1]
```diff
@@ -2095,6 +2115,43 @@ class Document final : public DocumentManagerInterface
         sendTextFrame(msg);
     }
 
+public:
+    void dumpState(std::ostream& oss)
+    {
+        oss << "Kit Document:\n"
+            << "\n\tstop: " << _stop
+            << "\n\tisLoading: " << _stop
```
Comment (Ashod) [inline kit/Kit.cpp:2123]: s/_stop/_isLoading/ ?

File: kit/Kit.cpp [hunk 2]
```diff
@@ -2243,8 +2326,33 @@ class KitSocketPoll : public SocketPoll
...
+
+    // unusual LOK event from another thread, push into our loop to process.
+    static bool pushToMainThread(LibreOfficeKitCallback callback, int type, const char *p, void *data)
+    {
+        if (mainPoll && mainPoll->getThreadOwner() != std::this_thread::get_id())
+        {
+            LOG_TRC("Unusual push callback to main thread");
+            std::shared_ptr<std::string> pCopy;
+            if (p)
+                pCopy = std::make_shared<std::string>(p, strlen(p));
```
Comment (Ashod) [inline kit/Kit.cpp:2338]: unique_ptr should do as well.

**Review by Ashod — APPROVED** [review]
There is a typo in one of the dump_state calls, otherwise looks good to me.

**Comment by mmeeks** [comment]
abandon =)

**Pull request closed without merging**